src/pages/workspaces/workspace/notebooks/TerminalLauncher.js

```diff
@@ -64,7 +64,8 @@ const TerminalLauncher = _.flow(
   }
 
   async componentDidUpdate() {
-    const { cluster: { clusterUrl } = {} } = this.props
+    const { cluster } = this.props || {}
+    const { clusterUrl } = cluster
```

Comment:
As of the latest change, this still exhibits the bug. I think the previous comment was based on a misreading of the code; providing a fallback value on the right would make sure the destructring always had an object to unpack, and `clusterUrl` would be undefined in the null case, which is good.

Reply:
Good catch. This wasn't the change I was thinking of, but I also didn't recognize that it would reintroduce the problem.

Emily and I chatted in person yesterday. My worry is that:
```
const { cluster } = this.props || {}
const { clusterUrl } = cluster || {}
```
doesn't express that we expect that `null` is a valid value for `cluster`... especially problematic because this code is distant from the origin of the `null` (`wrapWorkspace`). In other words, I might incorrectly simplify these two lines, effectively reverting the change. I would almost certainly miss that nuance if I was reading it in someone else's PR.

My suggestion in person (repeated here for broader communication) was to be explicit about checking for `null`.

Side note: This pattern appears in two places in this file... both here and when checking cluster status below. I thought, maybe we could have a utility function for this? Maybe lodash already has something useful? Answer: We could do something like:
```
const clusterUrl = _.get("cluster.clusterUrl", this.props)
```
but that also hides the fact that `null` is valid.

@panentheos, do you have any other thoughts/suggestions?

Reply:
In this case, I think `cluster || {}` communicates fairly well that `cluster` may be falsy, and so that should be accounted for in any refactors. This code path doesn't actually care about the `null` case, so I would hesitate to introduce any specific checks for that. Maybe a comment, if we think that would be valuable.